### PR TITLE
[rllib/tune] Add test for fractional gpu support in xray mode; add rllib support for fractional gpu

### PR DESCRIPTION
**Pull request by ericl**
@richardliaw should we also document it in tune?

Issues: https://github.com/ray-project/ray/issues/2593 https://github.com/ray-project/ray/issues/402

### CODE REVIEW

File: doc/source/rllib-training.rst [hunk 1]
```diff
@@ -50,6 +50,11 @@ In an example below, we train A2C by specifying 8 workers through the config fla
     python ray/python/ray/rllib/train.py --env=PongDeterministic-v4 \
         --run=A2C --config '{"num_workers": 8, "monitor": true}'
 
+Specifying Resources
+~~~~~~~~~~~~~~~~~~~~
+
+You can control the degree of parallelism used by setting the ``num_workers`` hyperparameter for most agents. Many agents also provide a ``num_gpus`` or ``gpu`` option. In addition, you can allocate a fraction of a GPU by setting ``gpu_fraction: f``. For example, with DQN you can pack five agents onto one GPU by setting ``gpu_fraction: 0.2``. Note that fractional GPU support requires enabling the experimental Xray backend by setting ``RAY_USE_XRAY=1``.
```
Comment (richardliaw) [inline doc/source/rllib-training.rst]: should note that `RAY_USE_XRAY` is a shell environment variable.

**Review by richardliaw — APPROVED** [review]

File: doc/source/rllib-training.rst (hunk 1 above)
Comment (ericl) [reply doc/source/rllib-training.rst]: Did so

**Comment by robertnishihara** [comment]
@richardliaw @ericl any feedback on fractional GPU support in xray? Is it intuitive? And behaving as expected?

**Comment by richardliaw** [comment]
I haven't actually tried out fractional GPU support for Tune; will do so now.

**Comment by ericl** [comment]
Test failure looks to be known.

**Pull request merged**